### PR TITLE
#2572, added a clause for default vhost in pid_of/1

### PR DESCRIPTION
**Pull request by ariel-anieli**
## Proposed Changes

- added a clause for default vhost in `pid_of/1`
- re-formatting in _rabbit_amqqueue.erl_
- implemented second half of `lookup/1`: clause with input as list was specified, but not done.

## Types of Changes

- [x] Bug fix (non-breaking change which fixes issue #2572)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [x] Cosmetic change (whitespace, formatting, etc)
- [ ] Build system and/or CI

## Checklist

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] All tests pass locally with my changes
- [ ] If relevant, I have added necessary documentation to https://github.com/rabbitmq/rabbitmq-website
- [ ] If relevant, I have added this change to the first version(s) in release-notes that I expect to introduce it

## Further Comments

None.

### CODE REVIEW

File: deps/rabbit/src/rabbit_amqqueue.erl [hunk 1]
```diff
@@ -339,7 +339,9 @@ is_server_named_allowed(Args) ->
             [amqqueue:amqqueue()].
 
 lookup(Name) when is_record(Name, resource) ->
-    rabbit_db_queue:get(Name).
+    rabbit_db_queue:get(Name);
+lookup(Names) when is_list(Names) ->
+    lists:map(fun rabbit_db_queue:get/1, Names).
```
Comment (michaelklishin) [inline deps/rabbit/src/rabbit_amqqueue.erl:344]: For better or worse, Team RabbitMQ has historically used list comprehensions over `lists:map/2` for basic map operations

File: deps/rabbit/src/rabbit_amqqueue.erl [hunk 2]
```diff
@@ -1578,11 +1580,13 @@ pid_of(Q) -> amqqueue:get_pid(Q).
...
 
-pid_of(VHost, QueueName) ->
-  case lookup(rabbit_misc:r(VHost, queue, QueueName)) of
+pid_of(VHost, QueueName) when is_binary(VHost) ->
+  case lookup(VHost, QueueName) of
     {ok, Q}                -> pid_of(Q);
     {error, not_found} = E -> E
-  end.
+  end;
+pid_of(default_vhost, QueueName) ->
```
Comment (michaelklishin) [inline deps/rabbit/src/rabbit_amqqueue.erl:1588]: This clause does not make much sense for an Erlang shell user. Please turn it into a `pid_of/1`.

File: deps/rabbit/src/rabbit_amqqueue.erl (hunk 1 above)
Comment (ariel-anieli) [reply deps/rabbit/src/rabbit_amqqueue.erl]: :slightly_smiling_face: So be it, @michaelklishin.

File: deps/rabbit/src/rabbit_amqqueue.erl (hunk 2 above)
Comment (ariel-anieli) [reply deps/rabbit/src/rabbit_amqqueue.erl:1588]: I have done it; thanks for your feedback.

Comment (michaelklishin) [reply deps/rabbit/src/rabbit_amqqueue.erl:1588]: My comment was not actually addressed.

`pid_of/2` with an artificial special case first argument does not make much sense.

``` erl
pid_of(QueueName) when is_binary(QueueName) ->
    VHost = rabbit_vhost:default_name(),
    case lookup(VHost, QueueName) of
    	{ok, Q}                -> pid_of(Q);
	    {error, not_found} = E -> E
    end.
```

If the virtual host is not explicitly given, and we only have a queue name, try looking it up in the default virtual host. If we are given a record, then it returns the pid value stored there.

Not very elegant but will break nothing and does not require us coming up with a clever short name for this new alternative to `pid_of/1` which accepts a record or `pid_of/2` which accepts a virtual host name.

**Review by michaelklishin — CHANGES_REQUESTED** [review]

File: deps/rabbit/src/rabbit_amqqueue.erl (hunk 2 above)
Comment (ariel-anieli) [reply deps/rabbit/src/rabbit_amqqueue.erl]: Thanks for this explanation, @michaelklishin; I did understand what you meant.

**Comment by ariel-anieli** [comment]
:slightly_smiling_face: Inception-like, but it does compile.

**Comment by ariel-anieli** [comment]
New push is about order of arguments in [`lookup/2`](https://github.com/rabbitmq/rabbitmq-server/blob/4c017a55d53feaf8bd369c58bfd180aeb7541f47/deps/rabbit/src/rabbit_amqqueue.erl#L358). It's now fixed.

**Comment by michaelklishin** [comment]
This PR has enough unrelated changes that I'd prefer to not accept it. Having an extra head of `pid_of/1` is a nice-to-have at best.

This modifies **super sensitive** (performance and otherwise) code paths for no reason related to #2572. In fact, it introduced a critical regression by swapping the argument order in `get_bcc_queue/2`.

While this was caught by Dialyzer, the gain here is not worth the risk.

**Comment by michaelklishin** [comment]
Thank you for taking the time to contribute. I'm afraid #2572 is not a good "new contributor issue" it seemed to be, in fact, I'd prefer to simply close it as a `wontfix`.

**Comment by ariel-anieli** [comment]
I see you're concerned about the changes done on `pid_of/1` & `get_bcc_queue/2`;
Thank you for this thorough feedback, @michaelklishin; I appreciate it.

**Comment by michaelklishin** [comment]
The changes around `pid_of/1` were harmless because it's a function invoked via an Erlang shell or `rabbitmqctl eval` only. Breaking them would be a annoyance at most.

**Comment by ariel-anieli** [comment]
Okay; I understand your point.

**Pull request closed without merging**